File: ios/VibeCoder/Models/Project.swift
import Foundation

struct Project: Identifiable, Codable {
    let id: String
    var title: String
    var description: String?
    var bundle: String? // base64 ZIP
    let creatorId: String
    var creatorName: String?
    var projectType: String
    var isPublic: Bool
    var playCount: Int
    var forkCount: Int
    var initialPrompt: String?
    var githubRepo: String?
    var freeTweaksRemaining: Int
    var publishedUrl: String?
    var parentProjectId: String?
    var createdAt: String?
    var updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, title, description, bundle
        case creatorId = "creator_id"
        case creatorName = "creator_name"
        case projectType = "project_type"
        case isPublic = "is_public"
        case playCount = "play_count"
        case forkCount = "fork_count"
        case initialPrompt = "initial_prompt"
        case githubRepo = "github_repo"
        case freeTweaksRemaining = "free_tweaks_remaining"
        case publishedUrl = "published_url"
        case parentProjectId = "parent_project_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct ProjectFile: Identifiable, Codable {
    var id: String { path }
    let path: String
    let size: Int

    var name: String {
        (path as NSString).lastPathComponent
    }

    var directory: String {
        let dir = (path as NSString).deletingLastPathComponent
        return dir.isEmpty ? "/" : dir
    }

    var fileExtension: String {
        (path as NSString).pathExtension.lowercased()
    }

    var iconName: String {
        switch fileExtension {
        case "html": return "doc.richtext"
        case "css": return "paintbrush"
        case "js": return "curlybraces"
        case "svg": return "photo"
        case "json": return "doc.text"
        default: return "doc"
        }
    }
}

struct ProjectVersion: Identifiable, Codable {
    var id: String { commitSha }
    let commitSha: String
    let message: String
    let date: String

    enum CodingKeys: String, CodingKey {
        case commitSha = "sha"
        case message, date
    }
}

struct ChatMessage: Identifiable {
    let id: String
    let role: MessageRole
    let content: String
    let timestamp: Date

    enum MessageRole {
        case user
        case assistant
        case system
    }

    init(role: MessageRole, content: String) {
        self.id = UUID().uuidString
        self.role = role
        self.content = content
        self.timestamp = Date()
    }
}

File: android/app/src/main/java/com/kreativekoala/vibecoder/data/model/Project.kt
package com.kreativekoala.vibecoder.data.model

import com.google.gson.annotations.SerializedName

data class Project(
    val id: String,
    val title: String,
    val description: String? = null,
    val bundle: String? = null,
    @SerializedName("creator_id") val creatorId: String,
    @SerializedName("creator_name") val creatorName: String? = null,
    @SerializedName("project_type") val projectType: String = "web_app",
    @SerializedName("is_public") val isPublic: Boolean = true,
    @SerializedName("is_featured") val isFeatured: Boolean = false,
    @SerializedName("play_count") val playCount: Int = 0,
    @SerializedName("view_count") val viewCount: Int = 0,
    @SerializedName("fork_count") val forkCount: Int = 0,
    val rating: Double = 0.0,
    @SerializedName("initial_prompt") val initialPrompt: String? = null,
    @SerializedName("github_repo") val githubRepo: String? = null,
    @SerializedName("tweak_count") val tweakCount: Int = 0,
    @SerializedName("free_tweaks_remaining") val freeTweaksRemaining: Int = 5,
    @SerializedName("published_url") val publishedUrl: String? = null,
    @SerializedName("preview_url") val previewUrl: String? = null,
    @SerializedName("thumbnail_url") val thumbnailUrl: String? = null,
    val status: String? = "ready",
    @SerializedName("parent_project_id") val parentProjectId: String? = null,
    @SerializedName("creation_method") val creationMethod: String = "ai_generated",
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("updated_at") val updatedAt: String? = null
)

data class ProjectFile(
    val name: String,
    val size: Int? = null,
    val type: String? = null,
    val content: String? = null
)
